app/models/models.js

```diff
@@ -2315,10 +2346,58 @@ YUI.add('juju-models', function(Y) {
         var relationData = endpoints.map(function(endpoint) {
           return endpoint[0] + ':' + endpoint[1].name;
         });
-        result.envExport.relations.push(relationData);
+        relations.push(relationData);
       }, this);
+      return relations;
+    },
 
-      return result;
+    /**
+      Generate a machine list for the exported yaml file based on the list of
+      machines passed in.
+
+      @method _generateMachineSpec
+      @param {Object} machineList The machines list.
+      @return {Object} The machine list for the export.
+    */
+    _generateMachineSpec: function(machineList) {
+      var machines = {};
+      var counter = 0;
+      machineList.each(function(machine) {
+        if (machine.parentId !== undefined) {
+          // We don't add containers to the machine spec.
+          return;
+        }
+        machines[counter] = {
+          series: machine.series,
+          constraints: this._collapseMachineConstraints(machine.hardware)
+        };
+        counter += 1;
+      }, this);
+      return machines;
+    },
+
+    /**
+      Collapses the machine hardware object details into the constraints string
+      expected by Juju.
+
+      @method _collapseMachineConstraints
+      @param {Object} constraints The harware constraints object from the
+        machine model.
+      @return {String} The constraints in a string format.
+    */
+    _collapseMachineConstraints: function(constraints) {
+      var constraint = '';
+      var constraintMap = {
+        arch: 'arch',
+        cpuCores: 'cpu-cores',
+        cpuPower: 'cpu-power',
+        mem: 'mem',
+        disk: 'root-disk'
```

Comment:
is this defined else where? I worry this is a corner case where we dupe this list and have to maintain it.